### PR TITLE
Tests: ensure consistency of transparency log response and entry

### LINKED ISSUE #184: Staging tests: assert that transparency log entry matches what's expected
**Description**

@asraa pointed out to me that in at least one instance, the transparency log entry that corresponds to the index we received during one of our staging tests does not have the same certificate that the signing event generated:

https://github.com/sigstore/sigstore-python/runs/7813004939?check_suite_focus=true has the following:

```
Using ephemeral certificate:
-----BEGIN CERTIFICATE-----
MIIFlDCCBRmgAwIBAgIUNNuunFTumt5pp7UpL91o+L2wq5MwCgYIKoZIzj0EAwMw
NzEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MR4wHAYDVQQDExVzaWdzdG9yZS1pbnRl
cm1lZGlhdGUwHhcNMjIwODEyMjAwNjA2WhcNMjIwODEyMjAxNjA2WjAAMHYwEAYH
KoZIzj0CAQYFK4EEACIDYgAEXkAlccAYYKzE96zfexkVJ0LGkDsvKldtxwnCS3O/
MGP2l4tEEhzdvpfV/44Syiln1eACQYE16QIo3ykJmsaUg7X6NDIHWOIbhf4hzH3X
LGZbNzUr3+Kl2LbYbPQYQUiWo4IEGzCCBBcwDgYDVR0PAQH/BAQDAgeAMBMGA1Ud
JQQMMAoGCCsGAQUFBwMDMB0GA1UdDgQWBBR+53uj/MVwBwBEZbJHgM66MdVg4zAf
BgNVHSMEGDAWgBRxhjCmFHxib/n31vQFGn9f/+tvrDBtBgNVHREBAf8EYzBhhl9o
dHRwczovL2dpdGh1Yi5jb20vc2lnc3RvcmUvc2lnc3RvcmUtcHl0aG9uLy5naXRo
dWIvd29ya2Zsb3dzL3N0YWdpbmctdGVzdHMueW1sQHJlZnMvaGVhZHMvbWFpbjA5
BgorBgEEAYO/MAEBBCtodHRwczovL3Rva2VuLmFjdGlvbnMuZ2l0aHVidXNlcmNv
bnRlbnQuY29tMBYGCisGAQQBg78wAQIECHNjaGVkdWxlMDYGCisGAQQBg78wAQME
KDE5YmY5OGYxZTU5NjkwYzlkZGM3NzZkMDFjYWM4Y2E5MTEwZmU3YmQwJAYKKwYB
BAGDvzABBAQWU3RhZ2luZyBJbnN0YW5jZSBUZXN0czAmBgorBgEEAYO/MAEFBBhz
aWdzdG9yZS9zaWdzdG9yZS1weXRob24wHQYKKwYBBAGDvzABBgQPcmVmcy9oZWFk
cy9tYWluMIICRwYKKwYBBAHWeQIEAgSCAjcEggIzAjECLwAbfBQqTpkrp98eH8V0
JFQTbBV6TLMcQQilIbixq+e/TAAAAYKTqfm0AAAEAQIAby7u8gEcNSPpLYQ/6PVv
9NukLRdMOzE8JXggxOv5ooHOYLsJ0dFIBkJZA5x7oeFYcpV66xJGcGNRATnAvbCv
qM4G9PywwzJNwxvS555jvlF6EjQDHmQwD4EGJieEjs0z3oYdlKmhp3GmE0xxhk7d
Feo1R3VKPSBi4wVpk+G1lDlyJm5iHwfYerGMMe3JC5pDhcBvQsRwMDSctoSrrXQT
kEIgqf5Ccf9GU2+0oJFVfrLg5DGT6Cr1bWKPevAmFAP5nZg8vW7bNEeL3jq3sO+l
VBsvq9/THyqaAMZGgEDRmK4owErSLyYZbyqgBOLypkVOzcWp/UYhRvcBAWFRLuCE
rK8thRIlU6eDYDFqN0NAgChpURLmPVdX+hUmjOrd2XSU5SHorwIfuG7nmJ+t/Mrk
mswbAE9/gIaXM6NlNfhzAL5Ld1r3VjW+AMMVwMD6ts5rdzQLcaieTMZi0Ke1blLh
tCxo9WCzdR8DGcFwM79DZxLIOJWoIu62ccW4bDKm7/SwbLyIJrsj/OvBfVRyk3WL
f/2f3wf6epFOr5r/WQNEB5Qom9vlbRtMVBT6Rw7Rc7wTRZl4cG20px6wPAD0ovtc
jLCe5oR/biQXHozedins0Qw7HVXpTNL4F6f40XptxDNpJI1FIcQ6adzt8L8uvEUH
wraiFSSxPkNdQzj8vdIrYqEwCgYIKoZIzj0EAwMDaQAwZgIxAIem1jPz9yebkMyH
ZpWFphACmcjqYeCsbNc9CVWCgrxgnKwfAtO7MSMVMejIAhT8lgIxAKXTzicmvrKD
k5geU+crlNSRKiiXRetfM8iOOZi6DnCaoE8dJkoiggORj9Hu7G6xCA==
-----END CERTIFICATE-----

Transparency log entry created at index: 7[90](https://github.com/sigstore/sigstore-python/runs/7813004939?check_suite_focus=true#step:4:91)0
Signature written to file README.md.sig
Certificate written to file README.md.crt
OK: README.md
```

But:

```
$ rekor-cli get --log-index 276 --rekor_server https://rekor.sigstage.dev --format json | jq ."Body".HashedRekordObj.signature.publicKey.content -r | base64 -d
-----BEGIN CERTIFICATE-----
MIICCTCCAa+gAwIBAgIGAYFO0EUEMAoGCCqGSM49BAMCMCoxDTALBgNVBAMMBHRl
c3QxGTAXBgNVBAoMEHRlc3QgY2VydGlmaWNhdGUwHhcNMjIwNjEwMTgxMTIwWhcN
MjIwNjEwMTgzMTIwWjAqMQ0wCwYDVQQDDAR0ZXN0MRkwFwYDVQQKDBB0ZXN0IGNl
cnRpZmljYXRlMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEL3clo6pNYhpxBeIk
7Ml3ti6338cMfE2CrnYuPdPdN0LEkIYd6ZaCFZACYMnR6DeZcyNU3FUgzT+qTGHK
KgHioqOBwDCBvTAdBgNVHQ4EFgQUWToDsTpiWgyfRHPnNRZuBIBqgaUwHwYDVR0j
BBgwFoAUsnR6usEbQjxDpkyta/b0VcegizowDgYDVR0PAQH/BAQDAgeAMBMGA1Ud
JQQMMAoGCCsGAQUFBwMDMAwGA1UdEwEB/wQCMAAwGwYDVR0RAQH/BBEwD4ENdGVz
dEB0ZXN0LmNvbTArBgorBgEEAYO/MAEBBB1odHRwczovL2Zha2VhY2NvdW50cy50
ZXN0LmNvbTAKBggqhkjOPQQDAgNIADBFAiB3lMwTlSSFIMYPIDVuvA1Zu1OeBg2J
YQxowlygrLkVEwIhALTfOukSkFbeRCbXU4m+ikpEocjPYLu9eW/PFBoQrgjG
-----END CERTIFICATE-----
```

These tests should probably also make some assertions in addition to the signing event:

- that the signing certificate matches
- that the signature matches
- that the entry type is correct
- perhaps other things?

### PR DESCRIPTION
**Pull request by woodruffw**
~~WIP.~~

Fixes #184.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ on:
 
 jobs:
   test:
+    permissions:
+      # Needed to access the workflow's OIDC identity.
+      id-token: write
```
Comment (woodruffw) [inline .github/workflows/ci.yml:15]: N.B.: I added this so that the `test_sign_rekor_entry_consistent` test can use the GitHub ambient OIDC credentials.

Comment (di) [reply .github/workflows/ci.yml:15]: Do we need additional markers to mark tests as 'interactive'? I would expect these to be somewhat annoying when trying to run the test suite locally without ambient credentials.

File: test/test_sign.py [hunk 2]
```diff
@@ -24,3 +28,23 @@ def test_signer_production():
...
+    result = signer.sign(payload, token)
+
+    expected_entry = result.log_entry
+    actual_entry = signer._rekor.log.entries.get(log_index=expected_entry.log_index)
+
+    assert expected_entry.uuid == actual_entry.uuid
+    assert expected_entry.body == actual_entry.body
+    assert expected_entry.integrated_time == actual_entry.integrated_time
+    assert expected_entry.log_id == actual_entry.log_id
+    assert expected_entry.log_index == actual_entry.log_index
```
Comment (di) [inline test/test_sign.py:50]:
```suggestion
    expected_entry = signer.sign(payload, token).log_entry
    actual_entry = signer._rekor.log.entries.get(log_index=expected_entry.log_index)

    assert expected_entry.uuid == actual_entry.uuid
    assert expected_entry.body == actual_entry.body
    assert expected_entry.integrated_time == actual_entry.integrated_time
    assert expected_entry.log_id == actual_entry.log_id
    assert expected_entry.log_index == actual_entry.log_index
```

Just a bit of simplification.

**Review by di — APPROVED** [review]
Fine as is, just some nits

File: .github/workflows/ci.yml (hunk 1 above)
Comment (woodruffw) [reply .github/workflows/ci.yml:15]: Yeah, we could add an interactive marker. Right now I have it set up such that the tests are skipped locally if no ambient credential is found, but we could have it do the normal interactive credential discovery instead.

Comment (di) [reply .github/workflows/ci.yml:15]: Oh, that's fine as well, I wasn't aware that's the case.

**Pull request merged**